### ISSUE TITLE
[v2.2.3 bug] - Duplicate output when @rid doesn't exists

### ISSUE THREAD

**Issue by dastoori** [orientechnologies/orientjs#218]
I've recently updated `orientjs` to 2.2.3 and noticed that in some case it returns duplicate records.

## Example:

```sql
/* schema */
create class Team extends V;
create property Team.name String;

create class Player extends V;
create property Player.name String;

create class TeamHasPlayer extends E;

/* data */
create vertex Team set name = 'TeamA';
create vertex Player set name = 'Player1';
create vertex Player set name = 'Player2';

create edge TeamHasPlayer from (select from Team) to (select from Player);
```

Following queries return two records and both of them are the same (notice the `@rid` field):

```javascript
// match query
db.query(`
  Match { class: Team, as: team }.out("TeamHasPlayer") {as: player}
  return team.name as team, player.name as player
`).then(res => {
  console.log(res);
});

// output
[ { '@type': 'd',
    team: 'TeamA',
    player: 'Player2',
    '@rid': { [String: '#-1:-1'] cluster: -1, position: -1 },
    '@version': 0 },
  { '@type': 'd',
    team: 'TeamA',
    player: 'Player2',
    '@rid': { [String: '#-1:-1'] cluster: -1, position: -1 },
    '@version': 0 } ]
```

another example:

```javascript
// select query
db.select('expand(out("TeamHasPlayer").include("name"))')
  .from('Team')
  .all()
  .then(res => {
    console.log(res);
  });

// output
[ { '@type': 'd',
    name: 'Player2',
    '@rid': { [String: '#-1:-1'] cluster: -1, position: -1 },
    '@version': 0 },
  { '@type': 'd',
    name: 'Player2',
    '@rid': { [String: '#-1:-1'] cluster: -1, position: -1 },
    '@version': 0 } ]
```

When i add `@rid` to `include()`, it returns the correct output:

```javascript
// query
db.select('expand(out("TeamHasPlayer").include("@rid", "name"))')
  .from('Team')
  .all()
  .then(res => {
    console.log(res);
  });

// output
[ { '@class': 'Player',
    '@type': 'd',
    name: 'Player2',
    '@rid': { [String: '#42:0'] cluster: 42, position: 0 },
    '@version': 0 },
  { '@class': 'Player',
    '@type': 'd',
    name: 'Player1',
    '@rid': { [String: '#41:0'] cluster: 41, position: 0 },
    '@version': 0 } ]
```

The above queries are works perfectly in v2.2.2
*i'm using orientdb 2.2.12*

**Comment by luigidellaquila** [comment]
Hi @dastoori 

This seems to be a regression in the SQL executor, I'm checking it

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @dastoori 

I tried to replicate this problem on 2.2.12 but I could not manage to do it.

Do you have a sample dataset to replicate it?

Thanks

Luigi

**Comment by dastoori** [comment]
This is my sample dataset *(from my first comment)*:

```sql
/* schema */
create class Team extends V;
create property Team.name String;

create class Player extends V;
create property Player.name String;

create class TeamHasPlayer extends E;

/* data */
create vertex Team set name = 'TeamA';
create vertex Player set name = 'Player1';
create vertex Player set name = 'Player2';

create edge TeamHasPlayer from (select from Team) to (select from Player);
```

The query outputs has no problem in "OrientDB Studio" (2.2.7 and 2.2.12) or "OrientJS v2.2.2", but when i install `npm install orientjs@2.2.3` it returns duplicate records.

**Comment by luigidellaquila** [comment]
Ok, got it. 
Reassigning it to @maggiolo00 

Thanks

Luigi

**Comment by maggiolo00** [comment]
@dastoori 

thanks opening this issue i've fixed it
i've added also a regression test

**Comment by maggiolo00** [comment]
@dastoori 

i'm waiting for travis then i will release 2.2.4 on NPM

**Comment by maggiolo00** [comment]
@dastoori 

just released 2.2.4
